src/server/api/routers/assignment/update.tsx

```diff
@@ -0,0 +1,43 @@
+import { z } from "zod";
+import { protectedProcedure } from "@/server/api/trpc";
+import { TRPCError } from "@trpc/server";
+
+export const updateAssignment = protectedProcedure
+  .input(
+    z.object({
+      id: z.string(),
+      data: z.object({
+        title: z.string().optional(),
+        description: z.string().optional(),
+        type: z.enum(["INDIVIDUAL", "GROUP", "EXAM", "PROJECT", "PREREQUISITE"]).optional(),
+        status: z.enum(["DRAFT", "PUBLISHED", "ARCHIVED"]).optional(),
+        submissionType: z.enum(["TEXT", "FILE", "LINK", "CODE", "MIXED"]).optional(),
+        submissionInstructions: z.string().optional(),
+        pointsPossible: z.number().optional(),
+        gradingRubric: z.object({}).optional(),
+        releaseDate: z.date().optional(),
```

Comment:
`z.string().datetime()` is also a valid use here - https://zod.dev/?id=datetimes